core/services/blockheaderfeeder/validate.go

```diff
@@ -0,0 +1,109 @@
+package blockheaderfeeder
+
+import (
+	"time"
+
+	"github.com/pelletier/go-toml"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/smartcontractkit/chainlink/core/services/job"
+)
+
+// ValidatedSpec validates and converts the given toml string to a job.Job.
+func ValidatedSpec(tomlString string) (job.Job, error) {
```

Comment:
Is this a typical pattern? It looks more like a Job constructor that also does validation - would it make sense to separate the two?

Reply:
:shrug: seems like we do this a bunch already so nbd

Reply:
yeah I had the same thought also. It seemed that all jobs follow the same pattern